parsec/core/fs/storage/user_storage.py

```diff
@@ -3,34 +3,21 @@
 
 from contextlib import asynccontextmanager
 from pathlib import Path
-from typing import AsyncIterator, Dict, Set, Tuple, cast
+from typing import AsyncIterator
 
-from parsec.core.fs.exceptions import FSLocalMissError
-from parsec.core.fs.storage.local_database import LocalDatabase
-from parsec.core.fs.storage.manifest_storage import ManifestStorage
-from parsec.core.fs.storage.version import get_user_data_storage_db_path
+import trio
+
+from parsec._parsec import UserStorage as _SyncUserStorage
+from parsec._parsec import (
+    user_storage_non_speculative_init as sync_user_storage_non_speculative_init,
+)
 from parsec.core.types import EntryID, LocalDevice, LocalUserManifest
 
 
 async def user_storage_non_speculative_init(data_base_dir: Path, device: LocalDevice) -> None:
-    data_path = get_user_data_storage_db_path(data_base_dir, device)
-
-    # Local data storage service
-    async with LocalDatabase.run(data_path) as localdb:
-
-        # Manifest storage service
-        async with ManifestStorage.run(
-            device, localdb, device.user_manifest_id
-        ) as manifest_storage:
-
-            timestamp = device.timestamp()
-            manifest = LocalUserManifest.new_placeholder(
-                author=device.device_id,
-                id=device.user_manifest_id,
-                timestamp=timestamp,
-                speculative=False,
-            )
-            await manifest_storage.set_manifest(device.user_manifest_id, manifest)
+    return await trio.to_thread.run_sync(
```

Comment:
Rust binding should expose an async function instead, see what has been done for `TimeProvider::sleep` (i.e. the binding function returns a `FutureIntoCoroutine`)

Reply:
1. `UserStorage` & co aren't async because `diesel` isn't so I can't do that 
2. We have an epic to improve `fs/storage` so we better to that in it

Reply:
(besides I would be surprised using this to_thread for every operation involving the storage creates a noticeable performance overhead)

Reply:
As we discussed yesterday, we need to encapsulate the sqlite part away from the business logic (given we have to support web). So it seems the right move to have this encapsulation defining an async API, and to implement it as a simple thread-per-connection with message passing in the sqlite implementation (the web one relying on browser API, they are already async by default)